common/logger/src/struct_log.rs

```diff
@@ -472,9 +472,19 @@ impl TCPStructLogThread {
     /// Writes a log line into json_lines logstash format, which has a newline at the end
     fn write_log_line(stream: &mut TcpWriter, message: String) -> io::Result<()> {
         let message = message + "\n";
+        let bytes = message.as_bytes();
+
+        // Attempt to write the log up to NUM_SEND_RETRIES + 1, and then drop it
```

Comment:
might be good to state that write_all will acquire a new TcpStream on write error.

Reply:
True, might be good to point out.  TcpWriter was supposed to be an abstraction so you don't have to worry about it, but i'll mention it here.